### ISSUE TITLE
Require cycle in generic.ts

### ISSUE THREAD

**Issue by danielwinkler** [paarthenon/variant#11]
I get the following warning at startup.
Should be solvable by importing directly instead of `from '.'`.

```
Require cycle: node_modules/variant/lib/index.js -> node_modules/variant/lib/generic.js -> node_modules/variant/lib/index.js

Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
```

**Comment by paarthenon** [comment]
Hi @danielwinkler thanks for bringing this up!

I've fixed this in variant `2.1`.

If you *were* using generics this version brings a small change to how they work. You can see the new version in action here:

https://stackblitz.com/edit/react-ts-dcmrug?file=option.ts

The documentation has also been updated.

**Comment by danielwinkler** [comment]
Great, thanks for the fix!
